core/runtime/wavm/module.cpp

```diff
@@ -37,6 +41,17 @@ namespace kagome::runtime::wavm {
       return nullptr;
     }
 
+    auto imports = WAVM::Runtime::getModuleIR(module).memories.imports;
+    if (imports.empty()) {
+      logger->critical("Memory type not found in imports.");
```

Comment:
Waaaait, it's totally possible for a memory to not be imported, kusama does this.